server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -40,7 +40,7 @@
  */
 public class ServerResource implements ResourceHandler {
    private final InvocationHelper invocationHelper;
-   private static final ServerInfo SERVER_INFO = new ServerInfo();
+   private static final byte[] SERVER_INFO = String.format("{\"version\": \"%s\"}", Version.printVersion()).getBytes();
```

Comment:
For trivial formats like this I would just use String concatenation.

Reply:
I would keep this as is, unless there is a noticeable performance difference.  
Server info is a holder that happens to have only the version now, but the idea is to expand it in the future

Reply:
String.format is considerably slower than a StringBuilder (which concatenation is replaced with). However since this is a static final the performance is negligible.

And by "keep this as is" you mean to revert this change completely?

To be honest using reflection for our responses to me would be considerably slower than if we registered mappers, which is what Quarkus will need to do anyways.

Reply:
> And by "keep this as is" you mean to revert this change completely?

I mean, keep ```ServerInfo``` holder

Reply:
I've added a manual Serializer implementation to avoid reflection.